### PR TITLE
Spectator and Respawn

### PR DESCRIPTION
**Pull request by AACO**
- [x] Add potato spectate
- [x] Cleanup spectate
- [x] Add potato respawn
- [x] Add headers to respawn
- [x] Fix map to world issue on respawn (off by many meters)
- [x] Add help display with controls in spectate.

Note: spectate tested on dedicated server, but always gotta roll them dice in full multiplayer.

### CODE REVIEW

**Comment by AACO** [comment]
This is functionally done, I'm going to add a few small detail tweaks tomorrow but the review can begin now. God fucking speed @PabstMirror with 169 files and 6.6k lines it might take a bit to read through.

File: addons/spectate/Displays.hpp [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        class MapGroup: RscControlsGroupNoScrollbars {
+            idc = MAP_GROUP_IDC;
+
+            x = QUOTE(-0.8 * XFACTOR + XOFFSET);
+            y = QUOTE(-0.8 * YFACTOR + YOFFSET);
+            w = QUOTE(41.5962 * XFACTOR);
+            h = QUOTE(26.7 * YFACTOR);
+
+            class controls {
+                class Map: RscMapControl {
```
Comment (PabstMirror) [inline addons/spectate/Displays.hpp]: maps don't like being in control groups
seems to not be aligned at my resolution, and clicking moves camera to wrong spot
best to move out of the control group, and just add cg's s x/y manually

Comment (AACO) [inline addons/spectate/Displays.hpp]: That's how it's handled in BI's EG spectate, can you do a video of the behavior and post your resolution?

Comment (PabstMirror) [inline addons/spectate/Displays.hpp]: 1280x1024

**Comment by PabstMirror** [comment]
Right clicking to rotate camera blocks camera movement with WASD

**Comment by AACO** [comment]
It doesn't unless you have right click bound to a mapped control re: @zerithsmash

File: addons/spectate/Displays.hpp (hunk 1 above)
Comment (AACO) [inline addons/spectate/Displays.hpp]: Had to remove some of the fancy scaling effects too: http://imgur.com/a/HfFaM
Should be resolved by https://github.com/BourbonWarfare/POTATO/pull/120/commits/9835582519780038954130d8979d8f21e3982198

**Comment by zerithsmash** [comment]
@PabstMirror 

The WASD like @AACO said I ran into as well. I also ran into that issue in-game in getting anything to lock properly. Unbinding that old Alpha controlset cleared the issue up. I don't know how many others have that button still bound to lock target, but I know you and I are some of the few who still have control profiles from back in Arma 3's Alpha.

**Comment by PabstMirror** [comment]
`inputAction "lockTarget"` - I can check later but I bet RMB is bound to this for me

**Comment by zerithsmash** [comment]
I would bet so as well.

**Comment by Alex6511** [comment]
It is for me.

**Pull request merged**